analyses/cell-type-ewings/exploratory_analysis/05-cluster-exploration.Rmd

```diff
@@ -0,0 +1,626 @@
+      forcats::fct_lump_n(7, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel("All remaining cell types", after = Inf)
+  )
+```
+
+
+```{r, fig.height=10, fig.width=7}
+# get heatmap for each library showing cluster assignments vs. singler assignments
+names(sce_list) |>
+  purrr::map(\(id){
+    cluster_celltype_heatmap(
+      cluster_classification_df,
+      id
+    )
+  })
+```
+
+From these plots, it's obvious that K = 5 is over clustering and cell types are spread across a lot of clusters. 
+However, for the other values of K, there doesn't seem to be any striking difference that are obvious by eye.  
+
+## Marker gene expression across clusters 
+
+We will also look at the expression of marker genes across each cluster. 
+In these plots, each row shows the distribution of the specified marker genes in that cluster. 
+Each panel is labeled with the marker gene expression being plotted. 
+
+Based on the above results, we will only show these plots for a value of K = 25. 
+
+```{r, fig.height=10, fig.width=5}
+# plot to look at marker gene expression for all cell types across all clusters
+# choose k=25 based on previous stats
+names(sce_list) |>
+  purrr::map(\(id) {
+    plot_marker_genes(cluster_classification_df,
+      k_value = "k_25",
+      id
+    )
+  })
+```
+
+For both samples we see very clear separation between clusters that express tumor marker genes and each of the normal cell marker genes. 
+
+## Assign cell types to clusters 
+
+Finally, we will assign cell types to all cells in a cluster based on a cluster having high expression of a set of marker genes. 
+
+- Any clusters with high tumor marker gene expression will be labeled "tumor". 
+- Any clusters with high immune marker gene expression also correspond to clusters that are predominantly "macrophage" in the heatmaps, so they will be labeled as "macrophage". 
+- Any clusters with high endothelial marker gene expression also correspond to clusters that are predominantly "endothelial cell" in the heatmaps, so they will be labeled as "macrophage". 
+- There is one cluster in `SCPCL000822` that has high mesenchymal gene expression and is predominantly chondrocytes so we will label this cell as "chondrocytes". 
+- All other clusters with high mesenchymal gene expression appear to have a mix of chondrocytes, fibroblasts, and smooth muscle cells so we will just label these as "mesenchymal-like cells". 
+
+```{r}
+# get umap embeddings for plotting later
+umap_df <- umap_df |>
+  dplyr::select(library_id, barcodes, UMAP1, UMAP2) |>
+  dplyr::distinct()
+
+# assign clusters based on marker gene expression and cell types that are prominent in that cluster
+# use the jaccard matrices to help guide assignments
+cluster_umap_df <- cluster_classification_df |>
+  dplyr::left_join(umap_df, by = c("library_id", "barcodes")) |>
+  dplyr::filter(nn_param == "k_25") |>
+  dplyr::mutate(
```

Comment:
Somewhere it is worth noting (comments? bullets? you decide!) which clusters are _not_ being assigned. Looks like 10&11 for `SCPCL000822` and 9 for `SCPCL000824`.